### ISSUE TITLE
Trouble pushing the files to the server

### ISSUE THREAD

**Issue by CarrieKami** [MSathieu/SimpleVCS#23]
Hello. 

I finished going through the book on creating a simple vcs and decided to go through this repo and try implement a few of the other functions such as push and pull. I am using localhost as my server but I am having trouble pushing my files to it. It is reading the directory that I have made locally but it is not pushing to the server though I have defined a directory in my xampp htdocs folder. Would you know what the issue is?

**Comment by MSathieu** [comment]
Which version are you using and what output are you getting from the command?

**Comment by CarrieKami** [comment]
I am using the most recent. I was not getting an error so I printed all of the parts of the post http request. 

`Data to be sent 
0a3a08ebd5c5046d9946e68db12a3f2395a36d83b0829625f9b31e82c80261cf cGFja2FnZSBtYWluCgppbXBvcnQgKAoJImVycm9ycyIKCSJmbGFnIgoJImZtdCIKCSJsb2ciCgoJIi4uL2NtZC9icmFuY2giCgkiLi4vY21kL2NvbW1pdCIKCSIuLi9jbWQvY29uZmlnIgoJIi4uL2NtZC9pbml0aWFsaXNlIgoJIi4uL2NtZC9sb2ciCgkiLi4vY21kL3B1c2giCgkiLi4vY21kL3NlcnZlciIKCSIuLi9nb3V0aWxzIgopCgpmdW5jIG1haW4oKSB7CglmbGFnLlVzYWdlID0gdXNhZ2UKCXZhciBicmFuY2hWYXIsIHVzZXJuYW1lLCBwYXNzd29yZCBzdHJpbmcKCXZhciBwdWJsaWMgYm9vbAoJZmxhZy5TdHJpbmdWYXIoJmJyYW5jaFZhciwgImJyYW5jaCIsICJtYXN0ZXIiLCAiU3BlY2lmeSB0aGUgYnJhbmNoLiIpCglmbGFnLlN0cmluZ1ZhcigmdXNlcm5hbWUsICJ1c2VybmFtZSIsICIiLCAiVGhlIHVzZXJuYW1lIGZvciBwdWxsaW5nL3B1c2hpbmciKQoJZmxhZy5Cb29sVmFyKCZwdWJsaWMsICJwdWJsaWMtcHVsbCIsIGZhbHNlLCAiTWFrZSBwdWxsaW5nIGZyb20gc2VydmVyIHB1YmxpYyIpCglmbGFnLlN0cmluZ1ZhcigmcGFzc3dvcmQsICJwYXNzd29yZCIsICIiLCAiVGhlIHBhc3N3b3JkIGZvciBwdWxsaW5nL3B1c2hpbmciKQoJZmxhZy5QYXJzZSgpCgoJZXhlY3V0ZWRDb21tYW5kIDo9IGZsYWcuQXJnKDApCglpZiBleGVjdXRlZENvbW1hbmQgPT0gIiIgewoJCWZsYWcuVXNhZ2UoKQoJCXJldHVybgoJfQoJaWYgZXhlY3V0ZWRDb21tYW5kICE9ICJpbml0IiAmJiBleGVjdXRlZENvbW1hbmQgIT0gInNlcnZlciIgJiYgIWdvdXRpbHMuQ2hlY2tJZkV4aXN0cygiLnZjcyIpIHsKCQlsb2cuRmF0YWwoIm5vdCBpbml0aWFsaXplZCIpCgl9Cgl2YXIgZXJyIGVycm9yCgoJc3dpdGNoIGV4ZWN1dGVkQ29tbWFuZCB7CgljYXNlICJpbml0IjoKCQlpZiBnb3V0aWxzLkNoZWNrSWZFeGlzdHMoIi52Y3MiKSB7CgkJCWxvZy5GYXRhbChlcnJvcnMuTmV3KCJhbHJlYWR5IGluaXRpYWxpc2VkIikpCgkJfQoJCWVyciA9IGluaXRpYWxpc2UuSW5pdGlhbGlzZShmbGFnLkFyZygxKSkKCWNhc2UgImNvbW1pdCI6CgkJZXJyID0gY29tbWl0LkNvbW1pdChmbGFnLkFyZygxKSkKCWNhc2UgImNvbmZpZyI6CgkJaWYgZmxhZy5BcmcoMikgPT0gIiIgewoJCQllcnIgPSBjb25maWcuR2V0Q29uZmlnKGZsYWcuQXJnKDEpKQoJCX0gZWxzZSB7CgkJCWVyciA9IGNvbmZpZy5TZXRDb25maWcoZmxhZy5BcmcoMSksIGZsYWcuQXJnKDIpKQoJCX0KCWNhc2UgImJyYW5jaCI6CgkJc3dpdGNoIGZsYWcuQXJnKDEpIHsKCQljYXNlICJjcmVhdGUiOgoJCQllcnIgPSBicmFuY2guQ3JlYXRlQnJhbmNoKGZsYWcuQXJnKDIpKQoJCWNhc2UgImRlbGV0ZSI6CgkJCWVyciA9IGJyYW5jaC5SZW1vdmVCcmFuY2goZmxhZy5BcmcoMikpCgkJY2FzZSAibGlzdCI6CgkJCWVyciA9IGJyYW5jaC5MaXN0QnJhbmNoZXMoKQoJCWRlZmF1bHQ6CgkJCWZtdC5QcmludGxuKCJJbnZhbGlkIGNvbW1hbmQiKQoJCX0KCWNhc2UgImxvZyI6CgkJZXJyID0gdmNzbG9nLkxvZyhicmFuY2hWYXIpCgljYXNlICJwdXNoIjoKCQllcnIgPSBwdXNoLlB1c2goZmxhZy5BcmcoMSksIHVzZXJuYW1lLCBwYXNzd29yZCkKCWNhc2UgInNlcnZlciI6CgkJZXJyID0gc2VydmVyLlNlcnZlcihwdWJsaWMpCglkZWZhdWx0OgoJCWZtdC5QcmludGxuKCJJbnZhbGlkIGNvbW1hbmQuIikKCX0KCWlmIGVyciAhPSBuaWwgewoJCWxvZy5GYXRhbChlcnIpCgl9Cn0KCi8vIHVzYWdlIGZ1bmN0aW9uIGFjdHMgbGlrZSBhIHNtYWxsIGRvYyBzdGF0aW5nIGFsbCBjb21tYW5kcyBhbmQgY2FuIGJlIHVzZWQgdG8gdGVzdCB0aGUgdGVybWluYWwKZnVuYyB1c2FnZSgpIHsKCWZtdC5QcmludGxuKCJDb21tYW5kczoiKQoJZm10LlByaW50bG4oIlx0Q29tbWl0LiIpCglmbXQuUHJpbnRsbigiXHRsb2c6IExvZ3MgYWxsIGNvbW1pdHMuIikKCWZtdC5QcmludGxuKCJBcmd1bWVudHM6IikKCWZsYWcuUHJpbnREZWZhdWx0cygpCn0K

headers
[USERNAME=root PASSWORD=1234]

End Request 
&{POST http://localhost/proj/files HTTP/1.1 1 1 map[Username:[root] Password:[1234]] {0xc0001121c0} 0x62a310 2682 [] false localhost map[] map[] <nil> map[]   <nil> <nil> <nil> <nil>}
err`

**Comment by MSathieu** [comment]
It is working fine for me, what do you mean with "I have defined a directory in my xampp htdocs folder". Are you using the SimpleVCS server?

**Comment by CarrieKami** [comment]
I am using localhost instead of the simpleVCS server.  Which is why I have added the directory for the vcs inside the htdocs folder since I use xampp to manage mySQL, apache etc for localhost. As I don't have access to a remote server I chose to simulate the push/pull and so on with localhost so that I could better learn and understand instead of using the SimpleVCS server.  Have you implemented an api on the SimpleVCS server to handle the requests from the vcs to the server?

**Comment by MSathieu** [comment]
You can't use apache for SimpleVCS, you must run the server locally with `simplevcs server`.

**Comment by MSathieu** [comment]
Yes indeed, if you run the server command for example in `~/simplevcs` and push to `localhost/test`, the files will be pushed to `~/simplevcs/test`. Hower in the directory you run the server, you need to put some files. in `auth.txt`, you need to put a list seperated with newlines of usernames and passwords (not hashed or encrypted) that can be used to push to the server. And you need a private and public certificate for the server. If you use a self signed certificate thats added to your keychain, you need to pass INSECURETLS=true to the push/pull command

**Comment by CarrieKami** [comment]
Ok. Thanks...I will do as you say and update you.

**Comment by CarrieKami** [comment]
I got the cerificates for the server and set up everything else . There are no more errors but now it silently fails. No files are pushed to the appropriate directory

**Comment by MSathieu** [comment]
Have you created a repository with `simplevcs init <name>` in the project subdirectory?
